bigquery/google/cloud/bigquery/dbapi/cursor.py

```diff
@@ -0,0 +1,102 @@
+#  limitations under the License.
+
+"""Cursor for the Google BigQuery DB-API."""
+
+import uuid
+
+from google.cloud.bigquery import job
+from google.cloud.bigquery.dbapi import _helpers
+
+
+class Cursor(object):
+    """Cursor to Google BigQuery.
+
+    :type: :class:`~google.cloud.bigquery.dbapi.Connection`
+    :param connection: A connection to Google BigQuery.
+    """
+    def __init__(self, connection):
+        self.connection = connection
+        self.description = None
+        self.rowcount = -1
+        self._query_results = None
+        self._page_token = None
+        self._has_fetched_all_rows = True
+
+    def close(self):
+        """No-op."""
+        pass
+
+    def _set_description(self, schema):
+        """Set description from schema."""
+        if schema is None:
+            self.description = None
+            return
+
+        desc = []
+        for field in schema:
+            desc.append(tuple([
+                field.name,
+                None,
+                None,
+                None,
+                None,
+                None,
+                field.mode == 'NULLABLE']))
+        self.description = tuple(desc)
+
+    def execute(self, operation):
+        """Prepare and execute a database operation."""
+        self._query_results = None
+        self._page_token = None
+        self._has_fetched_all_rows = False
+        client = self.connection._client
+        job_id = str(uuid.uuid4())
+        # TODO: parameters: if not ``None``, check if ``dict`` or sequence and
+        #       set up query parameters accordingly. Use values from
+        #       ``.setinputsizes()`` if it has been called, otherwise try to
+        #       infer types from parameter inputs.
+        query_job = client.run_async_query(job_id, operation)
+        query_job.use_legacy_sql = False
+        query_job.begin()
+        _helpers.wait_for_job(query_job)
+        self._query_results = query_job.results()
+        _, total_rows, _ = self._query_results.fetch_data(max_results=0)
+        # TODO: set rowcount to the number of DML affected rows if this was a
+        #       DML statement.
+        # https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2920
+        if total_rows is None:
+            total_rows = -1
+        self.rowcount = total_rows
+        self._set_description(self._query_results.schema)
+
+    def fetchone(self):
+        """Fetch a single row from the results of the last ``execute*()`` call.
+
+        :rtype: tuple
+        :returns:
+            A tuple representing a row or ``None`` if no more data is
+            available.
+        """
+        if self._has_fetched_all_rows:
+            return None
+
+        rows, _, page_token = self._query_results.fetch_data(
+                max_results=1, page_token=self._page_token)
+
+        if page_token is None:
+            self._has_fetched_all_rows = True
+
+        self._page_token = page_token
+        return rows[0]
```

Comment:
Also, seems like `fetchmany` and `fetchall` should be implemented, even in the first pass.